models.py

```diff
@@ -280,19 +280,38 @@ def __get_distance_weights(d, w1, w0, sigma):
     return weights
 
 
-def __get_size_weights(sizes):
-    C = Variable(torch.sqrt(torch.Tensor([sizes.size()[1] * sizes.size()[2]])) / 2, requires_grad=False)
-    if torch.cuda.is_available():
-        C = C.cuda()
+def __get_size_weights(sizes, C):
     size_weights = C / sizes
     size_weights[sizes == 1] = 1
     return size_weights
 
 
-def get_loss_params(w0, sigma):
+def get_loss_params(w0, sigma, imsize):
     w0 = Variable(torch.Tensor([w0]), requires_grad=False)
     sigma = Variable(torch.Tensor([sigma]), requires_grad=False)
+    C = Variable(torch.sqrt(torch.Tensor([imsize[0] * imsize[1]])) / 2, requires_grad=False)
     if torch.cuda.is_available():
         w0 = w0.cuda()
         sigma = sigma.cuda()
-    return {'w0': w0, 'sigma': sigma}
+        C = C.cuda()
+    return {'w0': w0, 'sigma': sigma, 'C': C}
+
+
+def mixed_dice_cross_entropy_loss(output, target, dice_weight, ce_weight, ce_loss=None):
+    dice_target = target[:, 0, :, :].long()
+    ce_target = target
```

Comment:
@apyskir I kinda get the idea of having ce and dice targets separate but it confuses me. Those are in fact the very same target and it would make it clearer if it was in fact just the `target` passed around

Reply:
@jakubczakon This formulation of dice_loss allows you to use it with just target matrix. For weighted cross entropy you need target matrix, distances and sizes and there is no possibility, that weighted cross entropy will work with simple target matrix. I can implement retrieving target matrix from `target` variable inside dice_loss, but then it will not be reusable to cases where you don't have these auxiliary targets.